### ISSUE TITLE
Create a unit test for the principal selection

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1542]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1538
- Created at 2012-09-25 14:30:01 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:20:54 as wontfix
- Assigned to nobody
---

We could write a unit test that would create a keytab with different principals using ktutil, and then attempt to select the correct principal using the selection algorithm in sss_krb5.c

The principal selection would be wrapped in a simple CLI program that would be executed from the unit test script which might be written in Python to ease interaction with ktutil.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-09-27 15:17:30](https://pagure.io/SSSD/sssd/issue/1538#221986)**

Fields changed

keywords:  => easyfix
milestone: NEEDS_TRIAGE => SSSD 1.10 beta
priority: major => minor
rhbz:  => 0
type: defect => task

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1538#221987)**

Fields changed

selected:  => Not need

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-01-02 15:32:09](https://pagure.io/SSSD/sssd/issue/1538#221988)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-07-30 12:20:45](https://pagure.io/SSSD/sssd/issue/1538#221989)**

Fields changed

changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
milestone: SSSD 1.13 beta => Interim Bucket
review:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-07-30 12:20:56](https://pagure.io/SSSD/sssd/issue/1538#221990)**

Fields changed

priority: minor => major

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-07-30 12:54:43](https://pagure.io/SSSD/sssd/issue/1538#221991)**

Fields changed

milestone: Interim Bucket => SSSD 1.12 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-02-19 11:52:09](https://pagure.io/SSSD/sssd/issue/1538#221992)**

Fields changed

blocking:  => #1923

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-05-30 16:12:24](https://pagure.io/SSSD/sssd/issue/1538#221993)**

Fields changed

milestone: SSSD 1.12 beta => SSSD 1.12.1

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-08 20:09:00](https://pagure.io/SSSD/sssd/issue/1538#221994)**

Mass-moving all tickets that didn't make 1.12.1 into 1.12.2

milestone: SSSD 1.12.1 => SSSD 1.12.2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-30 19:06:17](https://pagure.io/SSSD/sssd/issue/1538#221995)**

We need to do a release as requested by downstream. Moving tickets that are not fixed already or very close to acking to 1.12.3

milestone: SSSD 1.12.2 => SSSD 1.12.3

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-11-19 17:37:27](https://pagure.io/SSSD/sssd/issue/1538#221996)**

It makes sense to write the unit tests now instead of moving them to a later release again, but the tickets in 1.12.3 should be more focused at this point. Moving to 1.12.4

mark:  => 0
milestone: SSSD 1.12.3 => SSSD 1.12.4

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-01-14 11:54:56](https://pagure.io/SSSD/sssd/issue/1538#221997)**

Fields changed

owner: somebody => preichl

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-12 20:05:13](https://pagure.io/SSSD/sssd/issue/1538#221998)**

Fields changed

milestone: SSSD 1.12.4 => SSSD Deferred

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 14:24:50](https://pagure.io/SSSD/sssd/issue/1538#221999)**

**Metadata Update from @jhrozek**:
- Issue assigned to preichl
- Issue marked as blocked by: #1923
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-07-12 14:09:19](https://pagure.io/SSSD/sssd/issue/1538#448337)**

**Metadata Update from @jhrozek**:
- Assignee reset
- Custom field blocking reset (from #1923)
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue tagged with: easyfix

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:20:53](https://pagure.io/SSSD/sssd/issue/1538#636943)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:20:55](https://pagure.io/SSSD/sssd/issue/1538#636944)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
